### PR TITLE
Update ibm-cloud-provider-for-power-vs-platform to Include Persistent Storage implementation

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Updated Persistent Storage details in ibm-cloud-provider-for-power-vs-platform installer enhancement proposal.

### CODE REVIEW

**Comment by mkumatag** [comment]
/cc @jaypoulz @Prashanth684 @jsafrane 

please review the changes added for the CSI driver

/retitle Include Persistent Storage implementation

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md [hunk 1]
```diff
@@ -256,9 +256,26 @@ as because it is managing DNS & the Load Balancer (which is already set up with
...
+5. The ibm-powervs-block-csi-driver-operator itself is installed by [Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator) for all Openshift clusters in the PowerVS as per below steps
+   1. `cluster-version-operator` starts `openshift-cluster-storage-operator`
+   2. `openshift-cluster-storage-operator` checks if it runs on PowerVS, installs IBM PowerVS Block CSI driver Operator, starts it and monitors its status (i.e. monitors existence of the IBM PowerVS Block CSI driver Operator ClusterOperator CR and reports its own `openshift-cluster-storage-operator` status based on it)
+   3. IBM PowerVS Block CSI driver Operator starts and runs the IBM PowerVS Block CSI driver (i.e. starts Deployment with the controller parts and DaemonSet with the node parts). Additionally it creates at least one non-default StorageClass that users can use in their PVCs
+6. The ibm-powervs-block-csi-driver-operator leverages the PowerVS credentials by creating a credentials request to [Cluster Credential Operator](https://github.com/openshift/cloud-credential-operator).
```
Comment (Prashanth684) [inline enhancements/installer/ibm-cloud-for-power-vs-ipi.md:275]: nit: typo - Cloud Credential Operator

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md [hunk 2]
```diff
@@ -256,9 +256,26 @@ as because it is managing DNS & the Load Balancer (which is already set up with
...
+   1. A namespace called `openshift-cluster-csi-drivers`
+   2. Two ServiceAccounts: one for the Controller Service and other for the Node Service of the CSI driver
+   3. The RBAC rules to be used by the sidecar containers
+   4. The CSIDriver object representing the CSI driver
+   5. A Deployment that runs the driver's Controller Service
+   6. A DaemonSet that runs the driver's Node Service
+   7. A non-default StorageClass that uses the CSI driver as a provisioner.
+4. The ibm-powervs-block-csi-driver-operator deploys all the CSI driver objects in the namespace `openshift-cluster-csi-drivers`
+5. The ibm-powervs-block-csi-driver-operator itself is installed by [Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator) for all Openshift clusters in the PowerVS as per below steps
+   1. `cluster-version-operator` starts `openshift-cluster-storage-operator`
```
Comment (Prashanth684) [inline enhancements/installer/ibm-cloud-for-power-vs-ipi.md:272]: Are there any API changes involved with the cluster storage operator here?

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md (hunk 1 above)
Comment (Madhan-SWE) [reply enhancements/installer/ibm-cloud-for-power-vs-ipi.md]: fixed

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md (hunk 2 above)
Comment (Madhan-SWE) [reply enhancements/installer/ibm-cloud-for-power-vs-ipi.md:272]: We need to add the `ibm-powervs-block-csi-driver-operator` to the cluster storage operator as mentioned here (cluster storage operator section): https://docs.providers.openshift.org/container-storage-interface-driver/#deliverables

Comment (Madhan-SWE) [reply enhancements/installer/ibm-cloud-for-power-vs-ipi.md:272]: The following Openshift api changes are made: https://github.com/openshift/api/pull/1192/files

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md [hunk 3]
```diff
@@ -256,9 +256,27 @@ as because it is managing DNS & the Load Balancer (which is already set up with
...
+1. [ibm-powervs-block-csi-driver](https://github.com/kubernetes-sigs/ibm-powervs-block-csi-driver) is downstreamed to [openshift](https://github.com/openshift/ibm-powervs-block-csi-driver)
+2. The ibm-powervs-block-csi-driver-operator installs the IBM PowerVS Block CSI driver following the  [recommended guidelines](https://github.com/openshift/enhancements/blob/master/enhancements/storage/csi-driver-install.md)
+3. The ibm-powervs-block-csi-driver-operator deploys all the objects required by the  IBM PowerVS Block CSI driver:
+   1. A namespace called `openshift-cluster-csi-drivers`
+   2. Two ServiceAccounts: one for the Controller Service and other for the Node Service of the CSI driver
+   3. The RBAC rules to be used by the sidecar containers
+   4. The CSIDriver object representing the CSI driver
+   5. A Deployment that runs the driver's Controller Service
+   6. A DaemonSet that runs the driver's Node Service
+   7. A non-default StorageClass that uses the CSI driver as a provisioner.
```
Comment (jsafrane) [inline enhancements/installer/ibm-cloud-for-power-vs-ipi.md:269]: Why is the storage class non-default? Will there be other CSI driver that will provide the default one?

Comment (mkumatag) [reply enhancements/installer/ibm-cloud-for-power-vs-ipi.md:269]: May it is typo.. this will be a default storage class for the IPI deployment..

Comment (Madhan-SWE) [reply enhancements/installer/ibm-cloud-for-power-vs-ipi.md]: It was a typo. Fixed it now.

**Comment by dhellmann** [comment]
/retitle Update ibm-cloud-provider-for-power-vs-platform to Include Persistent Storage implementation

File: enhancements/installer/ibm-cloud-for-power-vs-ipi.md [hunk 4]
```diff
@@ -256,9 +256,27 @@ as because it is managing DNS & the Load Balancer (which is already set up with
...
+5. The ibm-powervs-block-csi-driver-operator itself is installed by [Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator) for all Openshift clusters in the PowerVS as per below steps
+   1. `cluster-version-operator` starts `openshift-cluster-storage-operator`
+   2. `openshift-cluster-storage-operator` checks if it runs on PowerVS, installs IBM PowerVS Block CSI driver Operator, starts it and monitors its status (i.e. monitors existence of the IBM PowerVS Block CSI driver Operator ClusterOperator CR and reports its own `openshift-cluster-storage-operator` status based on it)
+   3. An instance of `ClusterCSIDriver` will be created to faciliate managment of driver operator.  `ClusterCSIDriver` is already defined in - https://github.com/openshift/api/blob/master/operator/v1/types_csi_cluster_driver.go but needs to be expanded to include IBMPowerVSBlockCSIDriver
+   4. IBM PowerVS Block CSI driver Operator starts and runs the IBM PowerVS Block CSI driver (i.e. starts Deployment with the controller parts and DaemonSet with the node parts). Additionally it creates at least one non-default StorageClass that users can use in their PVCs
```
Comment (jsafrane) [inline enhancements/installer/ibm-cloud-for-power-vs-ipi.md:275]: "non-default" is still mentioned here

**Comment by jsafrane** [comment]
/lgtm
from storage POV

**Comment by jsafrane** [comment]
/approve
It has been on review for quite some time and nobody objected.

**Pull request merged**